internal/manifests/manifests.go

```diff
@@ -147,50 +131,86 @@ func (m *Manifests) DeleteClusterManifestInternal(ctx context.Context, params op
 	log := logutil.FromContext(ctx, m.log)
 	log.Infof("Deleting manifest from cluster %s", params.ClusterID.String())
 
-	// This call both verifies that the manifests are created for a valid cluster
-	// to align kube-api and ocm behavior, and get the cluster object to check that
-	// it is in valid state.
 	cluster, err := common.GetClusterFromDB(m.db, params.ClusterID, common.SkipEagerLoading)
 	if err != nil {
-		return common.NewApiError(http.StatusNotFound, fmt.Errorf("Object Not Found"))
+		err = fmt.Errorf("could not find cluster with id %s", params.ClusterID.String())
+		log.Infof(err.Error())
+		return common.NewApiError(http.StatusNotFound, err)
 	}
 
-	//Deletion of manifests is not allowed after installation has started.
-	preInstallationStates := []string{
-		models.ClusterStatusPendingForInput,
-		models.ClusterStatusInsufficient,
-		models.ClusterStatusReady,
+	err = m.validateAllowedToModifyManifests(ctx, cluster)
+	if err != nil {
+		log.WithError(err).Errorf("Not allowed to modify manifest for cluster id %s", params.ClusterID.String())
+		return err
 	}
-	if !funk.ContainsString(preInstallationStates, swag.StringValue(cluster.Status)) {
-		return common.NewApiError(http.StatusBadRequest, errors.Errorf("cluster %s is not in pre-installation states, "+
-			"can't remove manifests after installation has been started",
-			params.ClusterID.String()))
+
+	_, _, path := m.getManifestPathsFromParameters(ctx, params.Folder, &params.FileName)
+
+	err = m.deleteManifest(ctx, params.ClusterID, path)
+	if err != nil {
+		return err
 	}
 
-	if params.Folder == nil {
-		defaultFolder := models.CreateManifestParamsFolderManifests
-		params.Folder = &defaultFolder
+	log.Infof("Done deleting cluster manifest %s for cluster %s", path, params.ClusterID.String())
+	return nil
+}
+
+func (m *Manifests) UpdateClusterManifestInternal(ctx context.Context, params operations.V2UpdateClusterManifestParams) (*models.Manifest, error) {
+	if params.UpdateManifestParams.UpdatedFolder == nil {
+		params.UpdateManifestParams.UpdatedFolder = &params.UpdateManifestParams.Folder
+		params.UpdateManifestParams.UpdatedFileName = &params.UpdateManifestParams.FileName
 	}
```

Comment:
What if I only pass UpdatedFolder or only pass UpdatedFileName?  Both of these cases will break

Reply:
Please add unit tests for these cases